src/ifs/wl_seat/event_handling.rs

```diff
@@ -599,6 +619,52 @@ impl WlSeatGlobal {
         }
     }
 
+    fn touch_event(
+        self: &Rc<Self>,
+        dev: &DeviceHandlerData,
+        id: i32,
+        time_usec: u64,
+        event: TouchEvent,
+    ) {
+        self.state.for_each_seat_tester(|t| {
+            t.send_touch(self.id, time_usec, id, event);
+        });
+        match event {
+            TouchEvent::Down { pos } => {
+                let mapped_node = dev
+                    .output
+                    .get()
+                    .and_then(|out| out.node.get())
+                    .or_else(|| {
+                        self.state
+                            .builtin_output
+                            .get()
+                            .and_then(|con| self.state.root.outputs.get(&con))
+                    })
+                    .map(|o| o as Rc<dyn Node>)
+                    .unwrap_or_else(|| self.state.root.clone());
```

Comment:
This should use `DeviceHandlerData::get_rect`.

Reply:
I want it to fallback to `state.builtin_output` if there's no mapped output, but `get_rect` always returns a position. And the node is required anyways.

Reply:
I believe this should work identical to tablets. If there is a way to match an input device to an output device (possibly via the syspath, but this would have to be carefully analyzed), then this could serve as a default.

>And the node is required anyways.

I don't believe it is. See the equivalent code for tablets.